### PR TITLE
Cache responsibility sets

### PR DESCRIPTION
**Pull request by MartinEden**
https://vimc.myjetbrains.com/youtrack/issue/VIMC-360

### CODE REVIEW

File: app/src/main/contrib/models/ResponsibilitySet.ts [hunk 1]
```diff
@@ -48,4 +51,9 @@ export class ExtendedResponsibilitySet implements IExtendedResponsibilitySet {
         const responsibility = this.getResponsibilityByScenario(scenarioId);
         responsibility.coverageSets = coverageSets;
     }
+
+    getCoverageSets(scenarioId: string): CoverageSet[] {
+        const responsibility = this.getResponsibilityByScenario(scenarioId);
+        return responsibility.coverageSets;
+    }
```
Comment (littledice) [inline app/src/main/contrib/models/ResponsibilitySet.ts]: I'm a bit lost - do these changes relate to the caching story?

**Comment by MartinEden** [comment]
This should only be reviewed and merged once #8 and #9 are done.

What's going on here is that the responsibility store only stored one responsibility set at a time - the one you were currently looking at. In order to cache things I had to restructure the store so that it could store all the responsibility sets for all touchstones and all modelling groups.

File: app/src/main/contrib/stores/ResponsibilityStore.ts [hunk 2]
```diff
@@ -96,30 +112,37 @@ class ResponsibilityStore extends AbstractStore<ResponsibilityState, Responsibil
     }
 
     handleSetCurrentResponsibility(scenarioId: string) {
-        if (this.responsibilitySet != null) {
-            this.currentResponsibility = this.responsibilitySet.getResponsibilityByScenario(scenarioId);
+        const instance = this.getInstance();
+        const set = this.getInstance().getCurrentResponsibilitySet();
+        if (set != null) {
```
Comment (littledice) [inline app/src/main/contrib/stores/ResponsibilityStore.ts]: remove instance const

File: app/src/test/contrib/stores/ResponsibilityStoreTests.ts [hunk 3]
```diff
@@ -75,19 +74,24 @@ describe("ResponsibilityStore", () => {
         expect(responsibilityStore.getState().currentModellingGroup).to.eql(group);
     });
 
-    it("responsibilityActions.update sets responsibility set", () => {
+    it("responsibilityActions.update sets current responsibility set", () => {
```
Comment (littledice) [inline app/src/test/contrib/stores/ResponsibilityStoreTests.ts]: extra word

Comment (MartinEden) [inline app/src/test/contrib/stores/ResponsibilityStoreTests.ts]: Actually we were both confused here, but I think the code is right as it stands. We just confused the action name with the sentence.

**Comment by MartinEden** [comment]
All done here.

**Pull request merged**